src/test/java/org/minions/devfund/abel/moviedatabase/MovieDatabaseTest.java

```diff
@@ -0,0 +1,123 @@
+public class MovieDatabaseTest {
+
+    /**
+     * Test AddMovie.
+     */
+    @Test
+    public void testAddMovieNewMovieAndActors() {
+        MovieDatabase movieDatabase = new MovieDatabase();
+        String newMovie = "NewMovie";
+        movieDatabase.addMovie(newMovie, new String[]{"Actor1", "Actor2"});
+
+        assertEquals(1, movieDatabase.getMovieList().size());
+        assertEquals(newMovie, movieDatabase.getMovieList().get(0).getName());
+        assertEquals(2, movieDatabase.getActorList().size());
+    }
+
+    /**
+     * Test AddMovie.
+     */
+    @Test
+    public void testAddMovieAlreadyAdded() {
+        MovieDatabase movieDatabase = new MovieDatabase();
+        String newMovie = "NewMovie";
+        movieDatabase.addMovie(newMovie, new String[]{"Actor1", "Actor2"});
+        movieDatabase.addMovie(newMovie, new String[]{"Actor1", "Actor2"});
+
+        assertEquals(1, movieDatabase.getMovieList().size());
+        assertEquals(newMovie, movieDatabase.getMovieList().get(0).getName());
+        assertEquals(2, movieDatabase.getActorList().size());
+    }
+
+    /**
+     * Test AddMovie.
+     */
+    @Test
+    public void testAddMovieNewActors() {
+        MovieDatabase movieDatabase = new MovieDatabase();
+        String newMovie = "NewMovie";
+        String secondMovie = "SecondMovie";
+        movieDatabase.addMovie(newMovie, new String[]{"Actor1", "Actor2"});
+        movieDatabase.addMovie(secondMovie, new String[]{"Actor1", "Actor2", "Actor3"});
+        final int three = 3;
+
+        assertEquals(2, movieDatabase.getMovieList().size());
+        assertEquals(newMovie, movieDatabase.getMovieList().get(0).getName());
+        assertEquals(three, movieDatabase.getActorList().size());
+    }
+
+    /**
+     * Test AddRating.
+     */
+    @Test
+    public void testAddRatingNewMovie() {
+        MovieDatabase movieDatabase = new MovieDatabase();
+        String newMovie = "NewMovie";
+        movieDatabase.addMovie(newMovie, new String[]{"Actor1", "Actor2"});
+        final double ten = 10.0;
+        movieDatabase.addRating(newMovie, ten);
+
+        assertEquals(ten, movieDatabase.getMovieList().get(0).getRating(), 0.0);
+    }
+
+    /**
+     * Test UpdateRating.
+     */
+    @Test
+    public void testUpdateRatingToExistingMovie() {
+        MovieDatabase movieDatabase = new MovieDatabase();
+        String newMovie = "NewMovie";
+        movieDatabase.addMovie(newMovie, new String[]{"Actor1", "Actor2"});
+        final double ten = 10.0;
+        final double fifteen = 15.0;
+        movieDatabase.addRating(newMovie, ten);
+        movieDatabase.updateRating(newMovie, fifteen);
+
+        assertEquals(fifteen, movieDatabase.getMovieList().get(0).getRating(), 0.0);
+    }
+
+    /**
+     * Test GetBestActor.
+     */
+    @Test
+    public void testGetBestActor() {
+        MovieDatabase movieDatabase = new MovieDatabase();
+        String newMovie = "NewMovie";
+        String newMovie2 = "NewMovie2";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "NewMovie2" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)